source/jormungandr/jormungandr/street_network/geovelo.py

```diff
@@ -105,11 +105,45 @@ def _make_request_arguments_bike_details(cls, bike_speed_mps):
             'averageSpeed': bike_speed,  # in km/h, BEGINNER sets it to 13
         }
 
+    @classmethod
+    def _keep_obj_for_mode(cls, obj, modes):
+        # This filter is only necessary for tests without any physical_modes.
+        # In real life data physical_modes is never empty with depth = 2
+        if len(obj.stop_point.physical_modes) == 0:
+            return True
+        return any((mode.uri in modes for mode in obj.stop_point.physical_modes))
+
+    @classmethod
+    def _filter_bike_matrix_by_modes(cls, points):
+        # Filter matrix points with modes having priority order:
+        # Train', 'RapidTransit', 'Metro', 'Tramway', Car, Bus
+        train_modes = ['physical_mode:Train', 'physical_mode:RapidTransit']
+        metro_modes = ['physical_mode:Metro']
+        tram_modes = ['physical_mode:Tramway']
+        bus_modes = ['physical_mode:Car', 'physical_mode:Bus']
+        points_coord = []
+        for modes in (train_modes, metro_modes, tram_modes, bus_modes):
```

Comment:
sorry to bother you again... I think over this part of code... I think what you want is to order your points by the weight physical modes... you can improve the readability a little bit more by doing so...

Reply:
mode_weight = {"physical_mode:Train" : 1, 'physical_mode:RapidTransit': 2, 'physical_mode:Metro': 3 .... }
def key_func(point):
    return min((mode_weight[mode] for mode in obj.stop_point.physical_modes ))
return sorted(points, key=key_func)[:50]